docs/proposals/scheduledjob.md

```diff
@@ -216,6 +208,144 @@ In the above example:
+  multiple executions, which may overlap.
+- annotation with the nominal start time of the job, so that UIs and CLIs can show the
+  nominal start times of Jobs in relation to their parent, and so that the
+  controller can distinguish whether an existing job is started as
+  the most recent or a prior execution time (which may not be clear in the case
+  of controller-manager failures).  The annotation key will be 
+  `scheduledjob.kubernetes.io/nominal-start-time`.  The annotation value will be the RFC_3339
+  time and date of the intended start. 
+ - parent object will be specified using the standard created-by annotation, like pods of replica
+   sets.  The standard key is `kubernetes.io/created-by`.  The value is a serialized JSON object, like
+   `{ "kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ScheduledJob","namespace":"default",`
+   `"name":"nightly-earnings-report","uid":"5ef034e0-1890-11e6-8935-42010af0003e","apiVersion":...`
+   This serialization contains the UID of the parent.  This is used to match the Job to the SJ that created
+   it.
+
+## Updates to ScheduledJobs
+
+If the schedule were updated on a ScheduledJob it is hard for the controller to reason about
+whether a past job is intended to fulfill the current schedule or not.
+
+The controller matches Jobs to their ScheduledJob via the created-by annotation, which has a UID.
+If you delete a ScheduledJob and make an SJ with a the same name, the controller
+will not consider executions from the previous UID when making decisions about what executions to
+start, or status, etc.  When you replace, the old SJ is deleted, so you lose that history.
+So, if you want to preserve the old status, suspend the old one, and make one with a new name,
+or make a note of the old status.
+
+
+## Fault-Tolerance
+
+### Starting Jobs in the face of controller failures
+
+If the process with the scheduledJob controller in it fails,
+and takes a while to restart, the scheduledJob controller
+may miss the time window and it is too late to start a job.
+
+With a single scheduledJob controller process, we cannot give
+very strong assurances about not missing starting jobs.
+
+With a suggested HA configuration, there are multiple controller
+processes, and they use master election to determine which one
+is active at any time.
+
+If the Job's StartingDeadlineSeconds is long enough, and the
+lease for the master lock is short enough, and other controller
+processes are running, then a Job will be started.
+
+TODO: consider hard-coding the minimum StartingDeadlineSeconds
+at say 1 minute.  Then we can offer a clearer guarantee,
+assuming we know what the setting of the lock lease duration is.
+
+### Ensuring jobs are run at most once
+
+There are three problems here:
+
+- ensure at most one Job created per "start time" of a schedule.
+- ensure that at most one Pod is created per Job
+- ensure at most one container start occurs per Pod
```

Comment:
I don't think that's SJ controller problem, it should create a Job and delegate all the rest to the job controller.